src/core/meshcop/dataset_manager.cpp

```diff
@@ -638,6 +652,34 @@ Error DatasetManager::SendGetRequest(const Dataset::Components &aDatasetComponen
     return error;
 }
 
+void DatasetManager::CallMgmtSetCallback(Coap::Message &aMessage, Error aResult)
+{
+    Error    error;
+    StateTlv stateTlv;
+
+    SuccessOrExit(error = aResult);
+    VerifyOrExit(Tlv::FindTlv(aMessage, stateTlv) == kErrorNone, error = kErrorFailed);
+
+    switch (stateTlv.GetState())
+    {
+    case StateTlv::kReject:
+        error = kErrorRejected;
+        break;
+    case StateTlv::kAccept:
+        error = kErrorNone;
+        break;
+    default:
+        error = kErrorFailed;
+        break;
+    }
+
+exit:
+    otLogInfoMeshCoP("MGMT_SET finished: %s", ErrorToString(error));
```

Comment:
may not just logging for API call with callback, but consider always log the MGMT_SET response instead?

Reply:
Done.